tests/unit/s2n_security_policies_test.c

```diff
@@ -540,48 +541,77 @@ int main(int argc, char **argv)
     }
 
-    /* Test that security policy with invalid chacha20 boosting configuration triggers error on init */
+    /* Test security policy init with chacha20 boosting */
     {
-        /* Back up the first security policy selection because we will replace it with an invalid selection */
+        /* Back up the first security policy selection because the tests will replace it with a test selection */
         struct s2n_security_policy_selection previous = security_policy_selection[0];
 
-        struct s2n_cipher_suite *aes_128_only_cipher_suite_list[] = {
-            &s2n_tls13_aes_128_gcm_sha256,
-        };
-
-        struct s2n_cipher_preferences cipher_preferences = {
-            .count = s2n_array_len(aes_128_only_cipher_suite_list),
-            .suites = aes_128_only_cipher_suite_list,
-            .allow_chacha20_boosting = true
-        };
-
+        /* Setup cipher preferences + security policy */
+        struct s2n_cipher_preferences cipher_preferences = { 0 };
         struct s2n_security_policy test_policy  = {
             .minimum_protocol_version = S2N_SSLv3,
             .cipher_preferences = &cipher_preferences,
             .kem_preferences = &kem_preferences_null,
             .signature_preferences = &s2n_signature_preferences_20201021,
             .ecc_preferences = &s2n_ecc_preferences_test_all,
         };
-
         security_policy_selection[0] = (struct s2n_security_policy_selection){
             .version="test_security_policy_chacha20",
             .security_policy=&test_policy,
             .ecc_extension_required=0,
             .pq_kem_extension_required=0
         };
 
-        /* Cipher preferences has allow_chacha20_boosting incorrectly set as true even though the ciphersuite list only has aes128 */
+        /* Test security policy with invalid chacha20 boosting configuration triggers error on init */
         {
-            EXPECT_TRUE(cipher_preferences.allow_chacha20_boosting);
-            EXPECT_FAILURE_WITH_ERRNO(s2n_security_policies_init(), S2N_ERR_INVALID_SECURITY_POLICY);
+            struct s2n_cipher_suite *aes_128_only_cipher_suite_list[] = {
+                &s2n_tls13_aes_128_gcm_sha256,
+            };
+
+            cipher_preferences = (struct s2n_cipher_preferences) {
+                .count = s2n_array_len(aes_128_only_cipher_suite_list),
+                .suites = aes_128_only_cipher_suite_list,
+                .allow_chacha20_boosting = true
+            };
+
+            /* Cipher preferences has allow_chacha20_boosting incorrectly set as true even though the ciphersuite list only has aes128 */
+            {
+                cipher_preferences.allow_chacha20_boosting = true;
+                EXPECT_FAILURE_WITH_ERRNO(s2n_security_policies_init(), S2N_ERR_INVALID_SECURITY_POLICY);
+            }
+
+            /* Sanity check: cipher preferences has allow_chacha20_boosting correctly set as false */
+            {
+                /* cppcheck-suppress redundantAssignment */
```

Comment:
Why would this be a redundant assignment? I don't understand why cppcheck would be erroring.

Reply:
cppcheck doesn't think we actually used the previous assignment (when it was true). This was really annoying in the previous pr as well: https://github.com/aws/s2n-tls/pull/3543/files#diff-efe479836296a7e3f4e93f7cc220038045dbe850a4693e2e141b1a20ceec4bddR929.